File: app/javascript/channels/activity_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ActivityChannel", {
  connected() {
    this.perform("appear")
    console.log("connected to  ActivityChannel")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server

  },

  received(data) {
  	console.log(data);
    let elements = document.getElementsByClassName(`user-${data.user_id}-status`);
    
    window.elements = elements

    for(var i = 0; i < elements.length; i++){
    	elements[i].classList.remove('chat-online')
    	if(data.status == 'online')
    		elements[i].classList.add('chat-online')
    	
    		
    }
  }

 
});

File: app/channels/activity_channel.rb
class ActivityChannel < ApplicationCable::Channel
  def subscribed
    stream_from "activity_channel"
  end

  def unsubscribed
    ActionCable.server.broadcast "activity_channel", user_id: current_user.id, status: 'offline'
    current_user.online = false
  	current_user.save!
  end

  def appear
  	ActionCable.server.broadcast "activity_channel", user_id: current_user.id, status: 'online'
  	current_user.online = true
  	current_user.save!
  end

end
